packages/reference/src/common/EntityStore.tsx

```diff
@@ -0,0 +1,445 @@
+          entityType,
+          entityId,
+          (data: unknown) => {
+            queryClient.setQueryData(queryKey, data);
+          }
+        );
+      };
+      const queries: Query[] = queryCache.findAll({
+        type: 'active',
+        predicate: (query) => isSameSpaceEntityQueryKey(query.queryKey),
+      });
+      queries.forEach(subscribeQuery);
+
+      const unsubscribe = queryCache.subscribe((event) => {
+        if (!event) {
+          return;
+        }
+
+        const { type, query } = event;
+        const { queryKey, queryHash } = query;
+
+        if (!isSameSpaceEntityQueryKey(queryKey)) {
+          return;
+        }
+
+        if (type === 'added') {
+          subscribeQuery(query);
+        }
+
+        if (type === 'removed') {
+          // calling unsubscribe
+          entityChangeUnsubscribers.current[queryHash]?.();
+        }
+      });
+
+      return () => {
+        unsubscribe();
+        Object.values(entityChangeUnsubscribers.current).forEach((off) => off());
+        entityChangeUnsubscribers.current = {};
+      };
+    }, [onEntityChanged, queryCache, isSameSpaceEntityQueryKey, queryClient]);
+
+    return {
+      cmaClient,
+      fetch,
+      getResource,
+      getEntity,
+      ids: props.sdk.ids,
+      getEntityScheduledActions,
+    };
+  },
+  ({ fetch }) => fetch,
+  ({ getResource, getEntity, getEntityScheduledActions }) => ({
+    getResource,
+    getEntity,
+    getEntityScheduledActions,
+  }),
+  ({ ids }) => ids
+);
+
+export function useEntity<E extends FetchableEntity>(
+  entityType: FetchableEntityType,
+  entityId: string,
+  options?: UseEntityOptions
+): UseEntityResult<E> {
+  const { space, environment } = useCurrentIds();
+  const { getEntity } = useEntityLoader();
+  const queryKey: EntityQueryKey = [
+    entityType,
+    entityId,
+    options?.spaceId ?? space,
+    options?.environmentId ?? environment,
+  ];
+  const { status, data } = useQuery(queryKey, () => getEntity(entityType, entityId, options), {
+    enabled: options?.enabled,
+    useErrorBoundary: false,
+  });
+  return { status, data } as UseEntityResult<E>;
+}
+
+export function useResource(resourceType: ResourceType, urn: string, options?: UseResourceOptions) {
+  const queryKey: ResourceQueryKey = ['Resource', resourceType, urn];
+  const { getResource } = useEntityLoader();
+  const { status, data, error } = useQuery(
+    queryKey,
+    () => getResource(resourceType, urn, options),
+    {
+      enabled: options?.enabled,
+      // TODO: check if this is ok for all errors
+      useErrorBoundary: false,
+    }
+  );
+  return { status, data, error };
+}
+
+function EntityProvider({ children, ...props }: React.PropsWithChildren<EntityStoreProps>) {
+  const reactQueryClient = useMemo(() => {
+    const queryCache = new QueryCache();
+    const queryClient = new QueryClient({
+      queryCache,
+      defaultOptions: {
+        queries: {
+          useErrorBoundary: true,
```

Comment:
@z0al @aodhagan-cf can we change this to `false` please? It's on me to not pick the right default, but I'm certain we want the individual components to handle error cases and NOT escalate this to a potential non-existing error boundary.

Reply:
Yep I can do this in the follow up PR :)